client/src/partials/projects/project.js

```diff
@@ -1,117 +1,119 @@
-    }
-  };
-
-  dependencies.zs = {
-    query : {
-      tables : {
-        'mod_snis_zs' : {
-          columns : ['id', 'zone', 'territoire', 'province']
-        }
-      }
-    }
-  };
-
-  dependencies.enterprises = {
-    query : {
-      tables : {
-        'enterprise' : {
-          columns : ['id', 'name', 'abbr']
-        }
-      }
-    }
-  };
-
-  function handleErrors(err) {
-    messenger.danger('An Error occured : ' + err);
+function ProjectController(Projects, Enterprises, SnisService, StateFactory) {
+  var vm = this;
+
+  vm.enterprises = [];
+  vm.state = new StateFactory();
+  vm.view = 'default';
+
+  // bind methods
+  vm.create = create;
+  vm.update = update;
+  vm.cancel = cancel;
+  vm.submit = submit;
+  vm.del = del;
+
+  function handler(error) {
+    console.error(error);
+    vm.state.error();
   }
 
-  function load(models) {
-    for (var k in models) { $scope[k] = models[k]; }
+  // fired on startup
+  function startup() {
+    // load Projects
+    Projects.read(null, { complete : 1 })
+    .then(function (projects) {
+      vm.projects = projects.data;
+    }).catch(handler);
+
+    // load Enterprises
+    Enterprises.read().then(function (data) {
+      vm.enterprises = data;
+    }).catch(handler);
+
+    SnisService.healthZones().then(function (data) {
+      data.forEach(function (l) {
+        l.format = l.zone + ' - ' + l.territoire + ' (' + l.province + ')';
+      });
+      vm.zones = data;
+    }).catch(handler);
+
+    vm.state.reset();
   }
 
-  $scope.print = function print() {
-    $window.print();
-  };
-
-  $scope.new = function n() {
-    $scope.newProject = {};
-    $scope.action = 'new';
-  };
-
-  $scope.submitNew = function submitNew() {
-    var clean = connect.clean($scope.newProject);
-    connect.post('project', [clean])
-    .then(function (res) {
-      clean.id = res.data.insertId;
-      $scope.projects.post(clean);
-      $scope.action = 'default';
-    })
-    .catch(handleErrors);
-  };
-
-  $scope.cancelNew = function canceNew() {
-    $scope.newProject = {};
-  };
-
-  $scope.edit = function edit(project) {
-    $scope.editProject = angular.copy(project);
-    $scope.raw = project;
-    $scope.action = 'edit';
-  };
-
-  $scope.submitEdit = function submitEdit() {
-    var clean = connect.clean($scope.editProject);
-    connect.put('project', [clean], ['id'])
-    .then(function () {
-      $scope.projects.put(clean);
-      $scope.action = 'default';
-    })
-    .catch(handleErrors);
-  };
-
-  $scope.cancelEdit = function cancelEdit() {
-    $scope.editProject = angular.copy($scope.raw);
-    $scope.action = 'edit';
-  };
-
-  $scope.delete = function d(project) {
-    connect.delete('project', 'id', project.id)
-    .then(function () {
-      $scope.projects.remove(project.id);
+  function cancel() {
+    vm.state.reset();
+    vm.view = 'default';
+  }
+
+  function create() {
+    vm.view = 'create';
+    vm.project = {};
+  }
+
+  // Load a Project from the server
```

Comment:
The comment is not accurate.  it doesn't load any data from the server.  It simply binds data to the view.

... why does this need to be in a separate function?